lib/internal/http2/core.js

```diff
@@ -1683,41 +1686,20 @@ class Http2Stream extends Duplex {
     const req = createWriteWrap(this[kHandle], afterDoStreamWrite);
     req.stream = this[kID];
 
-    writeGeneric(this, req, data, encoding, cb);
+    if (writev)
+      writevGeneric(this, req, data, cb);
+    else
+      writeGeneric(this, req, data, encoding, cb);
 
     trackWriteState(this, req.bytes);
   }
 
   _writev(data, cb) {
-    // When the Http2Stream is first created, it is corked until the
-    // handle and the stream ID is assigned. However, if the user calls
-    // uncork() before that happens, the Duplex will attempt to pass
-    // writes through. Those need to be queued up here.
-    if (this.pending) {
-      this.once('ready', this._writev.bind(this, data, cb));
-      return;
-    }
-
-    // If the stream has been destroyed, there's nothing else we can do
-    // because the handle has been destroyed. This should only be an
-    // issue if a write occurs before the 'ready' event in the case where
-    // the duplex is uncorked before the stream is ready to go. In that
-    // case, drop the data on the floor. An error should have already been
-    // emitted.
-    if (this.destroyed)
-      return;
-
-    this[kUpdateTimer]();
-
-    if (!this.headersSent)
-      this[kProceed]();
-
-    var req = createWriteWrap(this[kHandle], afterDoStreamWrite);
-    req.stream = this[kID];
-
-    writevGeneric(this, req, data, cb);
+    this[kWriteGeneric](true, data, '', cb);
+  }
 
-    trackWriteState(this, req.bytes);
+  _write(data, encoding, cb) {
```

Comment:
Can we move this before `_writev` to keep the original order?

Reply:
Sure, will do.